infrastructure/src/main/java/org/corfudb/infrastructure/log/BoundedMap.java

```diff
@@ -18,7 +18,7 @@ public class BoundedMap {
     public BoundedMap(long offset, int size) {
         Preconditions.checkArgument(offset >= 0);
         Preconditions.checkArgument(size > 0);
-        Preconditions.checkArgument(Math.addExact(offset, size) < Integer.MAX_VALUE);
+        Preconditions.checkArgument(Math.addExact(offset, size) < Long.MAX_VALUE);
```

Comment:
What is the point of checking against `Long.MAX_VALUE`? `Math.addExact` will throw an exception in case of an overflow.

Reply:
You're right. Initially, I thought that the inequality needed to be strict but that doesn't seem required. Thank you - I've removed the `checkArgument` call and the test for this boundary.

Reply:
The point is readability and not functional. Maybe just add a comment.

I'm just worried that someone would remove it as "dead code" at a later point in time.